financial-services-accelerator/components/org.wso2.financial.services.accelerator.identity.extensions/src/main/java/org/wso2/financial/services/accelerator/identity/extensions/auth/extensions/request/validator/FSRequestObjectValidationExtension.java

```diff
@@ -155,4 +166,155 @@ protected boolean isRegulatory(OAuth2Parameters oAuth2Parameters) throws Request
+        final String clientId = jsonObject.getAsString(IdentityCommonConstants.CLIENT_ID);
+        String scope = jsonObject.getAsString(IdentityCommonConstants.SCOPE);
+        if (StringUtils.isBlank(clientId) || StringUtils.isBlank(scope)) {
+            log.error("Client id or scope cannot be empty");
+            throw new RequestObjectException("Client id or scope cannot be empty");
+        }
+        return true;
+    }
+
+    /**
+     * Validates scope.
+     *
+     * @param fsRequestObject
+     * @param dataMap
+     * @return
+     */
+    private String validateScope(FSRequestObject fsRequestObject, Map<String, Object> dataMap) {
+
+        try {
+            //remove scope claim
+            JWTClaimsSet claimsSet = fsRequestObject.getClaimsSet();
+            JSONObject claimsSetJsonObject = claimsSet.toJSONObject();
+            if (claimsSetJsonObject.containsKey("scope")) {
+                String scopeClaimString = claimsSetJsonObject.remove("scope").toString();
+                List allowedScopes = (List) dataMap.get("scope");
+                List<String> requestedScopes = new ArrayList<>(Arrays.asList(scopeClaimString.split(" ")));
+                StringBuilder stringBuilder = new StringBuilder();
+
+                // iterate through requested scopes and remove if not allowed
+                for (String scope : requestedScopes) {
+                    if (IdentityCommonConstants.OPENID_SCOPE.equals(scope)) {
+                        stringBuilder.append(scope).append(" ");
+                        log.debug("Adding Openid scope to the request object");
+                    }
+                    if (allowedScopes.contains(scope)) {
+                        stringBuilder.append(scope).append(" ");
+                        if (log.isDebugEnabled()) {
+                            log.debug(String.format("Adding allowed scope %s to the request object",
+                                    scope.replaceAll("[\r\n]", "")));
+                        }
+                    }
+                }
+                String modifiedScopeString = stringBuilder.toString().trim();
+                // throw an error if no valid scopes found or only one scope is found
+                if (StringUtils.isBlank(modifiedScopeString) || modifiedScopeString.split(" ").length <= 1) {
+                    throw new RequestObjectException("No valid scopes found in the request");
+                }
+                claimsSetJsonObject.put("scope", modifiedScopeString);
+                //Set claims set to request object
+                JWTClaimsSet validatedClaimsSet = JWTClaimsSet.parse(claimsSetJsonObject);
+                fsRequestObject.setClaimSet(validatedClaimsSet);
+                log.debug("Successfully set the modified claims-set to the request object");
+            }
+        } catch (ParseException | RequestObjectException e) {
+            return e.getMessage();
```

Comment:
Better to add an error log here.

Reply:
Fixed with 69b55aa0ba7b9206573f897b8ab47eaa78466fbe